nexus/src/app/background/driver.rs

```diff
@@ -88,17 +98,33 @@ impl Driver {
         imp: Box<dyn BackgroundTask>,
         opctx: OpContext,
         watchers: Vec<Box<dyn GenericWatcher>>,
-    ) -> TaskHandle {
+        activator: &Activator,
+    ) -> TaskName {
         // Activation of the background task happens in a separate tokio task.
         // Set up a channel so that tokio task can report status back to us.
         let (status_tx, status_rx) = watch::channel(TaskStatus {
             current: CurrentStatus::Idle,
             last: LastResult::NeverCompleted,
         });
 
-        // Set up a channel so that we can wake up the tokio task if somebody
-        // requests an explicit activation.
-        let notify = Arc::new(Notify::new());
+        // We'll use a `Notify` to wake up that tokio task when an activation is
+        // requested.  The caller provides their own Activator, which just
+        // provides a specific Notify for us to use here.  This allows them to
+        // have set up the activator before the task is actually created.
+        if let Err(previous) = activator.wired_up.compare_exchange(
+            false,
+            true,
+            Ordering::SeqCst,
+            Ordering::SeqCst,
```

Comment:
Candidly, I went with `SeqCst` as the safe choice for the reason you said -- it's not a hot path and it would take me some time to figure out a weaker but still safe choice here.